src/DepositManager.mo

```diff
@@ -151,44 +113,59 @@ module {
     /// - notifications are paused entirely
     /// This function never throws.
     public func notify(p : Principal) : async* ?(Nat, Nat) {
-      if (paused) return null;
-      let ?release = depositMap.obtainLock(p) else return null;
+      func do_notify() : async* ?(Nat, Nat) {
```

Comment:
Can we define this function outside of `notify()` before it? We just have to give it two arguments, `p` and `entry`. I think it would be easier to read.

It took me a minute to figure out what `entry` is inside this function. Had to realize that it is defined after this function. Didn't know that was possible in Motoko (function can use a variable from outside scope that is defined after).